pkg/network/vmispec/interface_test.go

```diff
@@ -101,6 +101,38 @@ var _ = Describe("VMI network spec", func() {
 		Entry("more then one interface", vmiSpecInterfaces(iface1, iface2, iface3), []string{iface1, iface2, iface3}),
 	)
 
+	Context("pop interface by network", func() {
+		const netName = "net1"
+		network := podNetwork(netName)
+		expectedStatusIface := v1.VirtualMachineInstanceNetworkInterface{Name: netName}
+		expectedStatusIfaces := vmiStatusInterfaces(iface1, iface2)
+
+		It("has no network", func() {
+			statusIface, _ := netvmispec.PopInterfaceByNetwork(vmiStatusInterfaces(iface1), nil)
+			Expect(statusIface).To(BeNil())
+		})
+
+		It("has no interfaces", func() {
+			statusIface, _ := netvmispec.PopInterfaceByNetwork(nil, &network)
+			Expect(statusIface).To(BeNil())
+		})
+
+		It("interface not found", func() {
+			statusIface, _ := netvmispec.PopInterfaceByNetwork(vmiStatusInterfaces(iface1, iface2), &network)
+			Expect(statusIface).To(BeNil())
+		})
+
+		DescribeTable("pop interface from position", func(statusIfaces []v1.VirtualMachineInstanceNetworkInterface) {
+			statusIface, statusIfaces := netvmispec.PopInterfaceByNetwork(statusIfaces, &network)
+			Expect(*statusIface).To(Equal(expectedStatusIface))
+			Expect(statusIfaces).To(Equal(expectedStatusIfaces))
+		},
+			Entry("first", vmiStatusInterfaces(netName, iface1, iface2)),
```

Comment:
How about generalizing this so it will also fit with the other 3 tests which have not been included in the table?

Something like this:
`Entry("first", vmiStatusInterfaces(netName, iface1, iface2), v1.VirtualMachineInstanceNetworkInterface{Name: netName}, vmiStatusInterfaces(iface1, iface2)),`

WDYT?

Reply:
I considered joining all the tests inside the table, however I didn't find a proper way (without having if statements) to handle .To(BeNil()) validations. If you have any ideas, I'll be happy to get some

Reply:
The snippet I posted earlier basically defined the table function signature with: The interface list, the network and the expected mutated list after the pop operation.
If you do that, there is no need to ask anything.

In the main body of the test table you will just do:
```
ifaceFound, updatedStatusIfaces := netvmispec.PopInterfaceByNetwork(statusIfaces, &network)
Expect(*ifaceFound).To(Equal(expectedStatusIface))
Expect(updatedStatusIfaces).To(Equal(expectedStatusIfaces))
```

Reply:
AFAIK, .To(Equal(expectedStatusIfaces)) is not supposed to handle nil validation, this what documentation says: 
func Equal(expected interface{}) GomegaMatcher
Equal uses reflect.DeepEqual to compare actual with expected. Equal is strict about types when performing comparisons. **It is an error for both actual and expected to be nil. Use BeNil() instead.**
That's the reason I didn't join it all inside the table.

Reply:
I did not know that one, thanks for pointing it out.

It could be resolved, using an `Or` matcher, but as it is not trivial, no point to block on it.